pandas/io/sas/sas.pyx

```diff
@@ -427,8 +427,10 @@ cdef class Parser:
                 jb += 1
             elif column_types[j] == column_type_string:
                 # string
-                string_chunk[js, current_row] = np.array(source[start:(
-                    start + lngt)]).tobytes().rstrip(b"\x00 ")
+                # Skip trailing whitespace
+                while lngt > 0 and source[start+lngt-1] in b"\x00 ":
```

Comment:
source[start+lngt-1] is a uint8_t isnt it?  am i wrong to think this looks weird?

Reply:
Read this as: check if the last byte is a null byte or white space.

Reply:
yah i get what it means, i just suspect there's an implicit casting going on in here

Reply:
What kind of casting? Actually I think it doesn’t matter because the both characters are on the positive side of `char`

Reply:
```c
        switch ((*((__pyx_t_6pandas_2io_3sas_4_sas_uint8_t const  *) ( /* dim=0 */ (__pyx_v_source.data + __pyx_t_17 * __pyx_v_source.strides[0]) )))) {
          case '\x00':
          case ' ':
          __pyx_t_8 = 1;
          break;
          default:
          __pyx_t_8 = 0;
          break;
        }
```

with

```c
typedef unsigned char __pyx_t_6pandas_2io_3sas_4_sas_uint8_t
```